### PR TITLE
Wallet deploy and test

### PR DESCRIPTION
**Pull request by dmtrskv**
Generating and deploying new wallet if one does not exist. First of all, required for tests (we have no other reasonable way to create a wallet in hardhat tests actually, because we must inject it into the handler context). But also can be useful for cli use, so that a user won't have to use nil_cli.

The private key is required by hardhat, so we shouldn't generate it.

Here I generate the wallet address if it is not provided.
If the wallet is not deployed to the address (either provided or generated), then it is deployed.

For this I fixed extendEnvironment which actually contained a race between the test and the setup (due to async func being called before setting up send and request functions; we should wait inside them on the promise).

The second commit introduces incrementer ignition module from example and a script running it. I think we'll need a trivial test of the hardhat cli of such kind.

### CODE REVIEW

File: src/utils/conversion.ts [hunk 1]
```diff
@@ -6,12 +9,16 @@ export function hexStringToUint8Array(hexString: string): Uint8Array {
   const byteArray = new Uint8Array(numBytes);
 
   for (let i = 0; i < numBytes; i++) {
-    byteArray[i] = Number.parseInt(cleanHexString.substr(i * 2, 2), 16);
+    byteArray[i] = Number.parseInt(cleanHexString.substring(i * 2, i * 2 + 2), 16);
```
Comment (dmtrskv) [inline src/utils/conversion.ts:12]: the former function is deprecated

**Comment by dmtrskv** [comment]
I do not get why build on 22 fails.

File: src/setup.ts [hunk 2]
```diff
@@ -54,34 +57,53 @@ export async function setupWalletAndClient(
...
-    throw new Error("Wallet address is not configured.");
+    walletAddress = bytesToHex(
+      WalletV1.calculateWalletAddress({
+        pubKey,
+        shardId: 1,
+        salt: newWalletSalt,
+      }),
+    );
+
+    console.log(
```
Comment (0xAleksaOpacic) [inline src/setup.ts:76]: If we deploy wallet for contract, i feel that we should at least print the wallet address, so in future, user can continue using this wallet

File: src/utils/conversion.ts (hunk 1 above)
Comment (0xAleksaOpacic) [reply src/utils/conversion.ts]: Thank you for improving this!

**Comment by 0xAleksaOpacic** [comment]
> I do not get why build on 22 fails.

Hey, just check the issue and it is related to node version.

To be more specific our GH action is using `v22.5.0`. And in the GH action you can see that installation failed:
https://github.com/NilFoundation/nil-hardhat-plugin/actions/runs/10009987332/job/27675560764?pr=19#step:3:50

I tried locally to use this version and have the same issue. I checked the latest release and found these is a minor fix and version `v22.5.1`. Tried locally using this version and no issues. 

If you checkout this release you can see that it fixes the error we have:  `Exit handler never called!` (https://nodejs.org/en/blog/release/v22.5.1)

So we should set Gh actions to use latest version (this one with patch)

File: src/setup.ts (hunk 2 above)
Comment (dmtrskv) [reply src/setup.ts:76]: but it's printed here

**Comment by dmtrskv** [comment]
Merged the commits and moved tests to the `tests` directory, so that they don't spam plugin structure with `contracts` and other hardhat dirs.

File: src/setup.ts (hunk 2 above)
Comment (0xAleksaOpacic) [reply src/setup.ts:76]: My bad... Friday afternoon...

**Review by 0xAleksaOpacic — APPROVED** [review]
Than you for opening PR and adding this!

**Pull request merged**